### PR TITLE
RavenDB-21996 LinqExtensions documentation

### PR DESCRIPTION
**Pull request by Lwiel**
### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-21996

### Additional description

Documentation of `LinqExtensions`

### Type of change

- [x] Bug fix
- [ ] Regression bug fix
- [ ] Optimization
- [ ] New feature

### How risky is the change?

- [x] Low 
- [ ] Moderate 
- [ ] High
- [ ] Not relevant

### Backward compatibility

- [x] Non breaking change
- [ ] Ensured. Please explain how has it been implemented?
- [ ] Breaking change
- [ ] Not relevant

### Is it platform specific issue?

- [ ] Yes. Please list the affected platforms.
- [x] No

### Documentation update

- [ ] This change requires a documentation update. Please mark the issue on YouTrack using `Documentation Required` tag.
- [x] No documentation update is needed 

### Testing by Contributor

- [ ] Tests have been added that prove the fix is effective or that the feature works
- [ ] Internal classes added to the test class (e.g. entity or index definition classes) have the lowest possible access modifier (preferable `private`) 
- [x] It has been verified by manual testing

### Testing by RavenDB QA team

- [ ] This change requires a special QA testing due to possible performance or resources usage implications (CPU, memory, IO). Please mark the issue on YouTrack using `QA Required` tag.
- [x] No special testing by RavenDB QA team is needed

### Is there any existing behavior change of other features due to this change?

- [ ] Yes. Please list the affected features/subsystems and provide appropriate explanation
- [x] No

### UI work

- [ ] It requires further work in the Studio. Please mark the issue on YouTrack using `Studio Required` tag.
- [x] No UI work is needed

### CODE REVIEW

File: src/Raven.Client/Documents/Session/IDocumentQueryBase.cs [hunk 1]
```diff
@@ -95,8 +95,6 @@ public interface IFilterDocumentQueryBase<T, TSelf> where TSelf : IDocumentQuery
         ///     Matches documents with value of chosen field matching searched terms.
         /// </summary>
         /// <param name="propertySelector">Property selector for the field that searched terms will be checked against.</param>
-        /// <param name="searchTerms">Space separated terms to search. If there is more than a single term, each of them will be checked independently.</param>
-        /// <param name="operator">Operator to be used for relationship between terms. Default: Or.</param>
```
Comment (arekpalinski) [inline src/Raven.Client/Documents/Session/IDocumentQueryBase.cs]: Why did it get removed?

File: src/Raven.Client/Documents/LinqExtensions.cs [hunk 2]
```diff
@@ -1044,17 +678,16 @@ public static IOrderedQueryable<T> ThenByScoreDescending<T>(this IOrderedQueryab
         }
 
         /// <summary>
-        /// Returns the query results as a stream
+        ///     Returns the query results as a stream.
```
Comment (arekpalinski) [inline src/Raven.Client/Documents/LinqExtensions.cs]: Why did you add `    ` (tab) here and in other places?

File: src/Raven.Client/Documents/Session/IDocumentQueryBase.cs (hunk 1 above)
Comment (Lwiel) [reply src/Raven.Client/Documents/Session/IDocumentQueryBase.cs]: Fixed

File: src/Raven.Client/Documents/LinqExtensions.cs (hunk 2 above)
Comment (Lwiel) [reply src/Raven.Client/Documents/LinqExtensions.cs]: I think it makes the summary more readable for documentation changing purposes

Comment (arekpalinski) [reply src/Raven.Client/Documents/LinqExtensions.cs]: The problem is that this introduces inconsistency.

Comment (Lwiel) [reply src/Raven.Client/Documents/LinqExtensions.cs]: Changed

**Review by arekpalinski — APPROVED** [review]
